library.c

```diff
@@ -1466,10 +1467,10 @@ PHP_REDIS_API int redis_sock_connect(RedisSock *redis_sock TSRMLS_DC)
         return -1;
     }
 
-    /* set TCP_NODELAY */
+    /* Attempt to set TCP_NODELAY if we're not using a unix socket */
     sock = (php_netstream_data_t*)redis_sock->stream->abstract;
-    if (setsockopt(sock->socket, IPPROTO_TCP, TCP_NODELAY, (char *) &tcp_flag, sizeof(int)) < 0) {
-        php_error_docref(NULL TSRMLS_CC, E_ERROR, "Can't activate TCP_NODELAY option!");
+    if (!usocket) {
+        setsockopt(sock->socket, IPPROTO_TCP, TCP_NODELAY, (char *) &tcp_flag, sizeof(int));
```

Comment:
Really?  Why does coverty insist on telling me what to do?  :smiley:

We could issue a warning (non fatal) which would depend on return type but I'm not sure if we need to.  How would you "not ignore" the return value if we're not going to do anything with it.